packages/client/src/client.ts

```diff
@@ -102,21 +107,141 @@ export class DefaultRoutingV1HttpApiClient implements RoutingV1HttpApiClient {
     }
   }
 
-  #mapProvider (event: RoutingV1HttpApiGetProvidersResponse): PeerInfo {
-    const peer = peerIdFromString(event.ID)
-    const ma: Multiaddr[] = []
+  async * getPeers (pid: PeerId, options: AbortOptions | undefined = {}): AsyncGenerator<Record, any, unknown> {
+    log('findPeers starts: %c', pid)
+
+    const signal = anySignal([this.shutDownController.signal, options.signal, AbortSignal.timeout(this.timeout)])
+    const onStart = defer()
+    const onFinish = defer()
+
+    void this.httpQueue.add(async () => {
+      onStart.resolve()
+      return onFinish.promise
+    })
+
+    try {
+      await onStart.promise
+
+      // https://specs.ipfs.tech/routing/http-routing-v1/
+      const resource = `${this.clientUrl}routing/v1/peers/${pid.toCID().toString()}`
+      const getOptions = { headers: { Accept: 'application/x-ndjson' }, signal }
+      const a = await fetch(resource, getOptions)
+
+      if (a.body == null) {
+        throw new CodeError('Routing response had no body', 'ERR_BAD_RESPONSE')
+      }
+
+      const contentType = a.headers.get('Content-Type')
+      if (contentType === 'application/json') {
+        const body = await a.json()
 
-    for (const strAddr of event.Addrs) {
-      const addr = multiaddr(strAddr)
-      ma.push(addr)
+        for (const peer of body.Peers) {
+          const record = this.#handleRecord(peer)
+          if (record !== null) {
+            yield record
+          }
+        }
+      } else {
+        for await (const peer of ndjson(toIt(a.body))) {
+          const record = this.#handleRecord(peer)
+          if (record !== null) {
+            yield record
+          }
+        }
+      }
+    } catch (err) {
+      log.error('findPeers errored:', err)
+    } finally {
+      signal.clear()
+      onFinish.resolve()
+      log('findPeers finished: %c', pid)
     }
+  }
+
+  async getIPNS (pid: PeerId, options: AbortOptions | undefined = {}): Promise<IPNSRecord> {
```

Comment:
Remove redundant `undefined`:

```suggestion
  async getIPNS (pid: PeerId, options: AbortOptions = {}): Promise<IPNSRecord> {
```